exchanges/deribit/deribit_wrapper.go

```diff
@@ -0,0 +1,1636 @@
+		return nil, err
+	}
+	var multiplier, contractSize float64
+	var settlementType futures.ContractSettlementType
+	for i := range contracts {
+		if !contracts[i].Name.Equal(fPair) {
+			continue
+		}
+		multiplier = contracts[i].Multiplier
+		settlementType = contracts[i].SettlementType
+	}
+
+	var baseSize float64
+	if r.Asset == asset.Futures {
+		baseSize = pos[index].SizeCurrency
+	} else if r.Asset == asset.Options {
+		baseSize = pos[index].Size
+	}
+	contractSize = multiplier * baseSize
+
+	return &futures.PositionSummary{
+		Pair:                      r.Pair,
+		Asset:                     r.Asset,
+		Currency:                  fPair.Base,
+		NotionalSize:              decimal.NewFromFloat(pos[index].MarkPrice),
+		Leverage:                  decimal.NewFromFloat(pos[index].Leverage),
+		InitialMarginRequirement:  decimal.NewFromFloat(pos[index].InitialMargin),
+		EstimatedLiquidationPrice: decimal.NewFromFloat(pos[index].EstimatedLiquidationPrice),
+		MarkPrice:                 decimal.NewFromFloat(pos[index].MarkPrice),
+		CurrentSize:               decimal.NewFromFloat(baseSize),
+		ContractSize:              decimal.NewFromFloat(contractSize),
+		ContractMultiplier:        decimal.NewFromFloat(multiplier),
+		ContractSettlementType:    settlementType,
+		AverageOpenPrice:          decimal.NewFromFloat(pos[index].AveragePrice),
+		UnrealisedPNL:             decimal.NewFromFloat(pos[index].TotalProfitLoss - pos[index].RealizedProfitLoss),
+		RealisedPNL:               decimal.NewFromFloat(pos[index].RealizedProfitLoss),
+		MaintenanceMarginFraction: decimal.NewFromFloat(pos[index].MaintenanceMargin),
+	}, nil
+}
+
+// GetOpenInterest returns the open interest rate for a given asset pair
+func (d *Deribit) GetOpenInterest(ctx context.Context, k ...key.PairAsset) ([]futures.OpenInterest, error) {
+	if len(k) == 0 {
+		return nil, fmt.Errorf("%w requires pair", common.ErrFunctionNotSupported)
+	}
+	for i := range k {
+		if k[i].Asset == asset.Spot ||
+			!d.SupportsAsset(k[i].Asset) {
+			return nil, fmt.Errorf("%w %v %v", asset.ErrNotSupported, k[i].Asset, k[i].Pair())
+		}
+	}
+	result := make([]futures.OpenInterest, len(k))
+	var err error
+	var pair currency.Pair
+	for i := range k {
+		pair, err = d.FormatExchangeCurrency(k[i].Pair(), k[i].Asset)
+		if err != nil {
+			return nil, err
+		}
+		var oi []BookSummaryData
+		if d.Websocket.IsConnected() {
+			oi, err = d.WSRetrieveBookBySummary(pair.Base.String(), d.GetAssetKind(k[i].Asset))
+		} else {
+			oi, err = d.GetBookSummaryByCurrency(ctx, pair.Base.String(), d.GetAssetKind(k[i].Asset))
+		}
+		if err != nil {
+			return nil, err
+		}
+		for a := range oi {
+			if oi[a].InstrumentName != pair.String() {
+				continue
+			}
+			result[i] = futures.OpenInterest{
+				Key: key.ExchangePairAsset{
+					Exchange: d.Name,
+					Base:     k[i].Base,
+					Quote:    k[i].Quote,
+					Asset:    k[i].Asset,
+				},
+				OpenInterest: oi[a].OpenInterest,
+			}
+			break
+		}
+	}
```

Comment:
```suggestion
	result := make([]futures.OpenInterest, 0, len(k))
	var err error
	var pair currency.Pair
	for i := range k {
		pair, err = d.FormatExchangeCurrency(k[i].Pair(), k[i].Asset)
		if err != nil {
			return nil, err
		}
		var oi []BookSummaryData
		if d.Websocket.IsConnected() {
			oi, err = d.WSRetrieveBookBySummary(pair.Base.String(), d.GetAssetKind(k[i].Asset))
		} else {
			oi, err = d.GetBookSummaryByCurrency(ctx, pair.Base.String(), d.GetAssetKind(k[i].Asset))
		}
		if err != nil {
			return nil, err
		}
		for a := range oi {
			if oi[a].InstrumentName != pair.String() {
				continue
			}
			result = append(result, futures.OpenInterest{
				Key: key.ExchangePairAsset{
					Exchange: d.Name,
					Base:     k[i].Base,
					Quote:    k[i].Quote,
					Asset:    k[i].Asset,
				},
				OpenInterest: oi[a].OpenInterest,
			})
			break
		}
	}
```

This will break `options` part of the test, but that's okay because it wasn't working - just didn't error. Please also update `options` part of the test to expect error (or make it unsupported)